dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo

```diff
@@ -48,17 +48,17 @@ equation
 
   // When the switch is closed, V and i are equal on both sides. Otherwise, the currents are zero.
   if (running.value) then
-    terminal1.V = terminal2.V;
     terminal1.i = - terminal2.i;
+    terminal1.V = terminal2.V;
     P1Pu = ComplexMath.real(terminal1.V * ComplexMath.conj(terminal1.i));
     Q1Pu = ComplexMath.imag(terminal1.V * ComplexMath.conj(terminal1.i));
     P1GenPu = - P1Pu;
     Q1GenPu = - Q1Pu;
     P2Pu = ComplexMath.real(terminal2.V * ComplexMath.conj(terminal2.i));
     Q2Pu = ComplexMath.imag(terminal2.V * ComplexMath.conj(terminal2.i));
   else
-    terminal1.i = Complex(0);
-    terminal2.i = Complex(0);
+    terminal1.i = terminal2.i;
+    terminal2.V = Complex(0);
```

Comment:
Could you remember me why we decided to change from the currents being zero to an equation with current equality and one with zero voltage?

Reading it again, it seems very strange for me, especially here in the generic model of an IdealSwitch as if I open a switch, it doesn't mean that I will have completely disconnected one of the two sides.

Reply:
When we investigated the load+transformer, here is what we observed:

- the transformer extvar are terminal1.V and terminal2.V

- in case running=true, we have equations linking the currents and voltages and everything is fine

- in case running=false, if we write terminal1.i=Complex(0) and terminal2.i=Complex(0), then there is no equation to compute terminal2.V

--> therefore we wrote terminal1.i=Complex(0) and terminal2.V=Complex(0)

For the gen+transformer+switch it is the same thing. But you're right that it is not generic or intuitive for the switch if it is used elsewhere (not the case for now but could be one day indeed...)

Reply:
In fact there it should be better to have terminal2.V=terminal1.V, that way it should still work I guess, and it would be more generic and understandable

Reply:
I'm not very at ease with this, I get the impression that the equations to write depend on the network handled (antenna or not).
But for me, as soon as you open a 2-port component, the only valid equations are i1 = i2 = 0 (if we neglegt the half-impedance and so on).